Configurations.md

```diff
@@ -0,0 +1,1760 @@
+{
+    // body
+}
+```
+
+#### `"PreferSameLine"`:
+
+```rust
+fn lorem() {
+    // body
+}
+
+fn lorem(ipsum: usize) {
+    // body
+}
+
+fn lorem<T>(ipsum: T)
+    where T: Add + Sub + Mul + Div {
+    // body
+}
+```
+
+#### `"SameLineWhere"`:
+
+```rust
+fn lorem() {
+    // body
+}
+
+fn lorem(ipsum: usize) {
+    // body
+}
+
+fn lorem<T>(ipsum: T)
+    where T: Add + Sub + Mul + Div
+{
+    // body
+}
+```
+
+## `fn_call_style`
+
+Indentation for function calls, etc.
+
+- **Default value**: `"Visual"`
+- **Possible values**: `"Block"`, `"Visual"`
+
+#### `"Block"`:
+
+```rust
+lorem(
+    "lorem",
+    "ipsum",
+    "dolor",
+    "sit",
+    "amet",
+    "consectetur",
+    "adipiscing",
+    "elit",
+);
+```
+
+#### `"Visual"`:
+
+```rust
+lorem("lorem",
+      "ipsum",
+      "dolor",
+      "sit",
+      "amet",
+      "consectetur",
+      "adipiscing",
+      "elit");
+```
+
+## `fn_call_width`
+
+Maximum width of the args of a function call before falling back to vertical formatting
+
+- **Default value**: `60`
+- **Possible values**: any positive integer
+
+#### Function call shorter than `fn_call_width`:
+```rust
+lorem("lorem", "ipsum", "dolor", "sit", "amet", "consectetur", "adipiscing", "elit");
+```
+
+#### Function call longer than `fn_call_width`:
+See [`fn_call_style`](#fn_call_style).
+
+## `fn_empty_single_line`
+
+Put empty-body functions on a single line
+
+- **Default value**: `true`
+- **Possible values**: `true`, `false`
+
+#### `false`:
+
+```rust
+fn lorem() {
+}
+```
+
+#### `true`:
+
+```rust
+fn lorem() {}
+```
+
+See also [`control_brace_style`](#control_brace_style).
+
+## `fn_return_indent`
+
+Location of return type in function declaration
+
+- **Default value**: `"WithArgs"`
+- **Possible values**: `"WithArgs"`, `"WithWhereClause"`
+
+#### `"WithArgs"`:
+
+```rust
+fn lorem(ipsum: Ipsum,
+         dolor: Dolor,
+         sit: Sit,
+         amet: Amet,
+         consectetur: Consectetur,
+         adipiscing: Adipiscing)
+         -> Elit
+    where Ipsum: Eq
+{
+    // body
+}
+
+```
+
+#### `"WithWhereClause"`:
+
+```rust
+fn lorem(ipsum: Ipsum,
+         dolor: Dolor,
+         sit: Sit,
+         amet: Amet,
+         consectetur: Consectetur,
+         adipiscing: Adipiscing)
+    -> Elit
+    where Ipsum: Eq
+{
+    // body
+}
+
+```
+
+## `fn_single_line`
+
+Put single-expression functions on a single line
+
+- **Default value**: `false`
+- **Possible values**: `true`, `false`
+
+#### `false`:
+
+```rust
+fn lorem() -> usize {
+    42
+}
+
+fn lorem() -> usize {
+    let ipsum = 42;
+    ipsum
+}
+```
+
+#### `true`:
+
+```rust
+fn lorem() -> usize { 42 }
+
+fn lorem() -> usize {
+    let ipsum = 42;
+    ipsum
+}
+```
+
+See also [`control_brace_style`](#control_brace_style).
+
+## `force_explicit_abi`
+
+Always print the abi for extern items
+
+- **Default value**: `true`
+- **Possible values**: `true`, `false`
+
+#### `false`:
+
+```rust
+extern {
+    pub static lorem: c_int;
+}
```

Comment:
Would be good to be explicit that non-"C" ABIs are always printed and that if false then "C" is removed (so 'force' in the name is a bit of a misnomer).